spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestPartitionedWrites.java

```diff
@@ -18,4 +18,23 @@
  */
 package org.apache.iceberg.spark.sql;
 
-public class TestPartitionedWrites extends PartitionedWritesTestBase {}
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import org.apache.iceberg.SnapshotSummary;
+import org.apache.iceberg.Table;
+import org.junit.jupiter.api.TestTemplate;
+
+public class TestPartitionedWrites extends PartitionedWritesTestBase {
+  @TestTemplate
+  public void testWapPropertiesNotSet() {
+    assertThatCode(
+            () -> {
+              sql("INSERT INTO %s VALUES (4, 'd')", tableName);
+              Table table = validationCatalog.loadTable(tableIdent);
+              assertThat(table.snapshot(table.refs().get("main").snapshotId()).summary())
+                  .doesNotContainKey(SnapshotSummary.WAP_BRANCH_PROP);
+            })
+        .doesNotThrowAnyException();
```

Comment:
Why do we need the outer `assertThatCode()...doesNotThrowAnyException()` method? 
Same for `TestPartitionedWritesToWapBranch`.